tomcat/src/main/java/com/linecorp/armeria/server/tomcat/TomcatService.java

```diff
@@ -487,16 +487,17 @@ private static void convertHeaders(HttpHeaders headers, MimeHeaders cHeaders) {
         }
     }
 
-    private static HttpHeaders convertResponse(Response coyoteRes) {
+    private static HttpHeaders convertResponse(Response coyoteRes, OutputBuffer outputBuffer) {
         final HttpHeaders headers = HttpHeaders.of(HttpStatus.valueOf(coyoteRes.getStatus()));
 
         final String contentType = coyoteRes.getContentType();
         if (contentType != null && !contentType.isEmpty()) {
             headers.set(HttpHeaderNames.CONTENT_TYPE, contentType);
         }
 
-        if (coyoteRes.getContentLengthLong() >= 0) {
-            headers.setLong(HttpHeaderNames.CONTENT_LENGTH, coyoteRes.getContentLengthLong());
+        final String method = coyoteRes.getRequest().method().toString();
+        if (!"HEAD".equals(method)) {
+            headers.setLong(HttpHeaderNames.CONTENT_LENGTH, outputBuffer.getBytesWritten());
```

Comment:
I found coyote response has getByteswritten as well, is this different? 

https://tomcat.apache.org/tomcat-5.5-doc/catalina/docs/api/org/apache/coyote/Response.html

Reply:
Ah, they are same. Simplified!